mlflow/pyspark/ml/__init__.py

```diff
@@ -101,37 +111,115 @@ def _get_estimator_info_tags(estimator):
     }
 
 
+def _is_parameter_search_estimator(instance):
+    from pyspark.ml.tuning import CrossValidator, TrainValidationSplit
+    return isinstance(instance, (CrossValidator, TrainValidationSplit))
+
+
+def _is_parameter_search_model(instance):
+    from pyspark.ml.tuning import CrossValidatorModel, TrainValidationSplitModel
+    return isinstance(instance, (CrossValidatorModel, TrainValidationSplitModel))
+
+
 def _get_instance_param_map(instance):
     from pyspark.ml.param import Params
-    from pyspark.ml.tuning import CrossValidator, TrainValidationSplit
 
     param_map = {
         param.name: instance.getOrDefault(param)
         for param in instance.params
         if instance.isDefined(param)
     }
     expanded_param_map = {}
+    is_tuning_estimator = _is_parameter_search_estimator(instance)
     for k, v in param_map.items():
-        if isinstance(v, Params):
+        if is_tuning_estimator and k == 'estimator':
+            expanded_param_map[k] = v.uid
+            # skip log estimator's nested params because they will be
+            # logged in nested runs.
+            # TODO:
+            #  consider the nested case:
+            #  the tuned estimator is a pipeline or a composite estimator like OneVsRest
+            #  then nested estimators uid need logged, e.g. uids of stage in
+            #  pipeline.stages/ova.classifier
```

Comment:
Discussion:
For this case, shall we log the hierarchy of the estimator ?
e.g. , CrossValidator tune on a pipeline, and pipeline includes many stages,
and the `estimatorParamMaps` include map of `Param --> value`, the Param here may be a param of a stage in the pipeline, and we log the param name as "{Param.parent}.{param_name}`, the `Param.parent` here is the uid of the stage(in the pipeline),

So log a hierarchy (includes all nested stage uids) for the tuned estimator is valuable.

Reply:
e.g. this example code:
https://github.com/apache/spark/blob/361444890e23057d32e737d88e8efa4fafde7ed3/examples/src/main/python/ml/cross_validator.py#L71

For the `lr.regParam` in autologging artifacts `estimatorParamMaps`, we will log an item of `{lr.uid}.regParam`

Reply:
Based on our discussion last week, I think it's okay to omit the parameters for the seed estimator, since we'll log the estimator parameters as part of each child run (as you've indicated in the inline comment above).

It definitely makes sense to log parameters for the optimal estimator after training, as discussed in https://github.com/mlflow/mlflow/pull/4270/files#r617115186, including parameters for constituent stages.

Reply:
I already updated PR to add the hierarchy if the estimator is a pipeline.